org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2017, 2021 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ui.tests.editor.quickfix;
+
+import java.util.Arrays;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.text.contentassist.ICompletionProposal;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
+import org.eclipse.xtext.ui.editor.XtextEditor;
+import org.eclipse.xtext.ui.testing.AbstractMultiQuickfixTest;
+import org.eclipse.xtext.ui.tests.quickfix.ui.tests.QuickfixCrossrefTestLanguageUiInjectorProvider;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+/**
+ * @author dhuebner - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(QuickfixCrossrefTestLanguageUiInjectorProvider.class)
+public class MultiQuickFixTest extends AbstractMultiQuickfixTest {
+	@Test
+	public void testFixMultipleMarkers() throws Exception {
+		String model =
+				"\"no doc\"\n" +
+				"Foo { ref Bor }\n" +
+				"\"no doc\" Bor { }\n";
+		IFile resource = dslFile(model);
+		IMarker[] markers = getMarkers(resource);
+		String contentAndMarkersBefore =
+				"<0<\"no doc\">0>\n" +
+				"Foo { ref Bor }\n" +
+				"<1<\"no doc\">1> Bor { }\n" +
+				"-----\n" +
+				"0: message=multiFixableIssue\n" +
+				"1: message=multiFixableIssue\n";
+		assertContentsAndMarkers(resource, markers, contentAndMarkersBefore);
+		applyQuickfixOnMultipleMarkers(markers);
+		String contentAndMarkersAfter =
+				"\"Better documentation\"\n" +
+				"Foo { ref Bor }\n" +
+				"\"Better documentation\" Bor { }\n" +
+				"-----\n" +
+				"(no markers found)\n";
+		assertContentsAndMarkers(resource, contentAndMarkersAfter);
+	}
+
+	@Test
+	public void testFixSingleMarker() throws Exception {
+		String model =
+				"\"no doc\"\n" +
+				"Foo { ref Bor }\n" +
+				"\"no doc\" Bor { }\n";
+		IFile resource = dslFile(model);
+		IMarker[] markers = getMarkers(resource);
+		String contentAndMarkersBefore =
+				"<0<\"no doc\">0>\n" +
+				"Foo { ref Bor }\n" +
+				"<1<\"no doc\">1> Bor { }\n" +
+				"-----\n" +
+				"0: message=multiFixableIssue\n" +
+				"1: message=multiFixableIssue\n";
+		assertContentsAndMarkers(resource, markers, contentAndMarkersBefore);
+		IMarker firstMarker = Iterables.getFirst(IterableExtensions.sortBy(Arrays.asList(markers), (IMarker it) -> {
```

Comment:
Code is repeated accross test files ... a shared helper method would be nice.

Reply:
done